packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp

```diff
@@ -2008,6 +2010,123 @@ TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL( Import_Util, UnpackAndCombineWithOwningPIDs,
 }
 
 
+// Unit Test the functionality in Tpetra_Import_Util
+TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL( Import_Util, SortCrsEntries, LO, GO, Scalar)  {
+
+  typedef typename Teuchos::Array<GO>::size_type size_type;
+
+  int max_num_entries_per_row = 7;  // should be odd
+  int num_cols = 15; // should be odd
+
+  Teuchos::Array<size_t>  rowptr, rowptr2;
+  Teuchos::Array<GO>  colind, colind2;
+  Teuchos::Array<Scalar> vals, vals2;
+
+  // Fill the CRS arrays, use random values
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_real_distribution<Scalar> dist(1.0, 2.0);
+  int row = 0;
+  while (true) {
+    int m = (max_num_entries_per_row - 1) / 2;
+    int start = std::max(0, row - m);
+    int num_cols_this_row = std::min(row + 1 + m,
+        std::min(max_num_entries_per_row, m+num_cols-row));
+    int end = start + num_cols_this_row;
+
+    rowptr.push_back(static_cast<size_t>(colind.size()));
+    rowptr2.push_back(static_cast<size_t>(colind2.size()));
+
+    for (int col=start; col<end; col++) {
+      colind.push_back(col);
+      colind2.push_back(col);
+
+      Scalar rands = dist(gen);
+      vals.push_back(rands);
+      vals2.push_back(rands);
+
+      // Create duplicate for merge test
+      colind2.push_back(col);
+      vals2.push_back(rands);
+    }
+
+    if (row > 1 && num_cols_this_row == 1 + m) break;
+    row++;
+
+  }
+  size_type num_entries = colind.size();
+  rowptr.push_back(static_cast<size_t>(num_entries));
+
+  size_type num_entries2 = colind2.size();
+  rowptr2.push_back(static_cast<size_t>(num_entries2));
+
+  // Randomly shuffle values and column indices
+  Teuchos::Array<Scalar>  vals_rand(num_entries);
+  Teuchos::Array<GO>  colind_rand1(num_entries), colind_rand2(num_entries);
+  for (size_t i=0; i < static_cast<size_t>(rowptr.size()-1); ++i) {
+    size_t num_cols_this_row = rowptr[i+1] - rowptr[i];
+    Teuchos::Array<GO> ix(num_cols_this_row);
+    std::iota(ix.begin(), ix.end(), rowptr[i]);
```

Comment:
Ah, `std::iota` got included in C++11:

http://en.cppreference.com/w/cpp/algorithm/iota

I was worried, because it was an SGI thing that didn't make it into C++98 and isn't present in all C++98 implementations, but since it's in C++11, it's OK to use :-D .